site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md

```diff
@@ -0,0 +1,395 @@
+ }
+```
+
+### Activate Python environment using conda :
+```shell
+conda activate nlp_ml_sentiment_analysis
+```
+
+### Run Streamlit App :
+```shell
+cd streamlit
+streamlit run Sentiment_Analysis_APP.py
+```
+
+The full code of the use case is also available in this Notebook **[Sentiment_Analysis_NLP_with_Snowpark_ML](https://github.com/Snowflake-Labs/snowpark-python-demos/blob/main/snowpark_nlp_ml_demo/notebook/Sentiment_Analysis_NLP_with_Snowpark_ML.ipynb)** 
+
+```shell
+cd notebook
+jupyter notebook
+```
+
+<!-- ------------------------ -->
+## Snowflake Environment Setup
+Duration: 2
+
+### Option 1 - Automatically : with Streamlit App
+Use the Streamlit App to setup Snowflake Objects
+
+![Puppy](./assets/Setup_Snowflake_Objects.png)
+
+Make sure you have this result :
+
+![Puppy](./assets/Setup_Snowflake_Objects_Results.png)
+
+You can check directly with Snowsight that the data are available in Snowflake.
+
+![Puppy](./assets/DB_IMDB.png)
+
+### Option 2 - Manually : with Snowsight
+First, log into your Snowflake Account (Snowsight Web UI) using your credentials.
+
+Then, run the following SQL commands to create the DATABASE :
+
+```sql
+USE ROLE ACCOUNTADMIN;
+
+CREATE DATABASE if not EXISTS IMDB;
+```
+
+Run the following SQL commands to create the TABLES :
+```sql
+USE DATABASE IMDB;
+USE SCHEMA PUBLIC;
+
+CREATE TABLE if not EXISTS TRAIN_DATASET (
+	REVIEW STRING,
+	SENTIMENT STRING
+);
+
+CREATE TABLE if not EXISTS TEST_DATASET (
+	REVIEW STRING,
+	SENTIMENT STRING
+);
+```
+
+Run the following SQL commands to create the WAREHOUSE :
+```sql
+CREATE WAREHOUSE if not EXISTS DEMO_WH WAREHOUSE_SIZE=MEDIUM INITIALLY_SUSPENDED=TRUE AUTO_SUSPEND=120;
+```
+
+Run the following SQL commands to create the STAGE :
+```sql
+CREATE STAGE if not EXISTS MODELS;
+
+USE IMDB.PUBLIC;
+```
+
+
+<!-- ------------------------ -->
+## Load Data
+Duration: 5
+
+We used Python code to load the data into Snowflake. In order to simplify code execution you can click on the right button to start loading the data.
+
+### What You'll Do :
+
+Use use the section **Load Data** :
+
+![Puppy](./assets/Load_data_Section.png)
+
+#### Step 1 : Load Train Dataset
+
+Here is the display that we expect after the execution.
+
+![Puppy](./assets/Step_1_Load_Train_Dataset.png)
+
+#### Step 2 : Load Test Dataset
+
+Here is the display that we expect after the execution.
+
+![Puppy](./assets/Step2_Load_Test_Dataset.png)
+
+### What You'll Learn :
+
+#### Load Data into Snowflake with Snowpark
+
+```shell
+with z.open("TRAIN_DATASET.csv") as f:
+    pandas_df = pd.read_csv(f)
+    session.write_pandas(pandas_df, "TRAIN_DATASET", auto_create_table=False, overwrite=True)
+```
+
+<!-- ------------------------ -->
+## Analyze Data
+Duration: 2
+
+### What You'll Do :
+
+Use use the section **Analyze** to explore and analyze the datasets and see some metrics.
+
+![Puppy](./assets/Analyze_Data1.png)
+
+#### Select your data
+Choose the dataset that you want to analyze :
```

Comment:
A space is not needed between the final word of the statement and `:`. Can you please update this in the entire file?

```suggestion
Choose the dataset that you want to analyze:
```